magpy/gui/dialogclasses.py

```diff
@@ -112,6 +115,172 @@ def DropFavs(self, e):
+
+        elemlist = [(self.baseLabel, noOptions),
+                emptySpace,
+                (self.baseTextCtrl, expandOption),
+                (self.obsIDLabel, noOptions),
+                (self.idComboBox, expandOption),
+                emptySpace,
+                (self.formatLabel, noOptions),
+                (self.formatComboBox, expandOption),
+                emptySpace,
+                (self.typeLabel, noOptions),
+                (self.typeComboBox, expandOption),
+                emptySpace,
+                (self.sampleLabel, noOptions),
+                (self.sampleTextCtrl, expandOption),
+                emptySpace,
+                (self.startTimeLabel, noOptions),
+                (self.startDatePicker, expandOption),
+                (self.startTimePicker, expandOption),
+                emptySpace,
+                (self.endTimeLabel, noOptions),
+                (self.endDatePicker, expandOption),
+                (self.endTimePicker, expandOption),
+                emptySpace,
+                (self.elementsLabel, noOptions),
+                (self.elementsTextCtrl, expandOption),
+                emptySpace,
+                (self.generatedLabel, noOptions),
+                (self.generatedTextCtrl, expandOption),
+                (self.okButton, dict(flag=wx.ALIGN_CENTER)),
+                (self.closeButton, dict(flag=wx.ALIGN_CENTER))]
+
+        # A GridSizer will contain the other controls:
+        cols = 1
+        rows = int(np.ceil(len(elemlist)/float(cols)))
+        gridSizer = wx.FlexGridSizer(rows=rows, cols=cols, vgap=5, hgap=10)
+
+        # Add the controls to the sizers:
+        for control, options in elemlist:
+            gridSizer.Add(control, **options)
+
+        for control, options in \
+                [(gridSizer, dict(border=5, flag=wx.ALL))]:
+            boxSizer.Add(control, **options)
+
+        self.SetSizerAndFit(boxSizer)
+
+    def bindControls(self):
+        self.baseTextCtrl.Bind(wx.EVT_TEXT, self.onChange)
+        self.startTimePicker.Bind(wx.EVT_TEXT, self.onChange)
+        self.endTimePicker.Bind(wx.EVT_TEXT, self.onChange)
+        self.idComboBox.Bind(wx.EVT_CHOICE, self.onChange)
+        self.formatComboBox.Bind(wx.EVT_CHOICE, self.onChange)
+        self.typeComboBox.Bind(wx.EVT_CHOICE, self.onChange)
+        self.sampleTextCtrl.Bind(wx.EVT_TEXT, self.onChange)
+        self.startDatePicker.Bind(wx.EVT_DATE_CHANGED, self.onChange)
+        self.startTimePicker.Bind(wx.EVT_TEXT, self.onChange)
+        self.endDatePicker.Bind(wx.EVT_DATE_CHANGED, self.onChange)
+        self.endTimePicker.Bind(wx.EVT_TEXT, self.onChange)
+        self.elementsTextCtrl.Bind(wx.EVT_TEXT, self.onChange)
+        self.generatedTextCtrl.Bind(wx.EVT_TEXT, self.onOverride)
+
+    def onChange(self, e):
+        self.createUrl()
+
+    def onOverride(self, e):
+        self.url = self.generatedTextCtrl.GetValue()
+
+    def createUrl(self):
+        stday = self.startDatePicker.GetValue()
```

Comment:
split this method into methods for each parameter so it's easier to follow, then this method can assemble the url

- getBaseUrl()
- getObservatory()
- getStartTime()
- getEndTime()
  possibly one method for parsing a datetime from the value of a date and time picker...
- getFormat()
- getElements()
- getDataType()
- getPeriod()